workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -1,32 +1,87 @@
 ﻿using Improbable.Gdk.Core;
-using Improbable.Gdk.Mobile;
 using UnityEngine;
-using UnityEngine.UI;
 
 namespace Playground
 {
+    [ExecuteInEditMode]
     public class ConnectionScreenController : MonoBehaviour
     {
         private const string HostIpPlayerPrefsKey = "SpatialOSHostIp";
+        private const int GuiPadding = 10;
+        private const float VerticalPositionRatio = 0.4f;
 
-        [SerializeField] private GameObject connectionPanel;
-        [SerializeField] private InputField ipAddressInput;
-        [SerializeField] private Button connectButton;
-        [SerializeField] private Text errorMessage;
         [SerializeField] private GameObject clientWorkerPrefab;
+        [SerializeField] private Font font;
+        [SerializeField] private float screenWidthFontRatio = 20;
 
-        private GameObject worker;
+        private bool showConnectionGui = true;
+        private bool isConnecting = false;
+        private string ipAddressText = null;
+        private string errorMessage = "";
 
-        private string IpAddress => ipAddressInput != null ? ipAddressInput.text : null;
+        private GameObject worker;
 
         public void Awake()
         {
-            ipAddressInput.text = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
-            connectButton.onClick.AddListener(TryConnect);
+            ipAddressText = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
+        }
+
+        public void OnGUI()
+        {
+            if (!showConnectionGui)
```

Comment:
how about just disabling the gameobject if we don't want to show it

Reply:
that way we could also get rid of the try-finally part

Reply:
Agreed, we can even destroy it 😁

Reply:
Removed bool, it now destroys the gameobject on success

Reply:
Tested this and it doesn't cause any problems as expected